.github/workflows/ci.yml

```diff
@@ -505,6 +500,13 @@ jobs:
       upgrade-to-newer-dependencies: ${{ needs.build-info.outputs.upgrade-to-newer-dependencies }}
       include-success-outputs: ${{ needs.build-info.outputs.include-success-outputs }}
       debug-resources: ${{ needs.build-info.outputs.debug-resources }}
+      is-special-tests-require: ${{
```

Comment:
Not sure if there is better way to handle this. openapi tests executes when `needs-api-codegen` is true and other special tests executes based on these conditions.
```
needs.build-info.outputs.run-tests == 'true' &&
   (needs.build-info.outputs.canary-run == 'true' ||
        needs.build-info.outputs.upgrade-to-newer-dependencies != 'false' ||
        needs.build-info.outputs.full-tests-needed == 'true')
```

To support both in special tests workflow, am sending the evaluated condition value. so inside special tests workflow we can do if check.

Reply:
That's a good idea. But NIT: we do not do `is-` prefix in other conditional parameter. Can we change it to just `special-tests-required` ?

Reply:
Hmm. After a bit of thinking. one bad side-effect it has is that the whole "special tests" group will be evaluated and shown in the tree of tests for every PR - now for most PRs the whole group is skipped. 

Maybe a better approach will be to move that one to "Static Checks, Mypy Docs" and rename it to something like "CI image checks"  (as the whole group?). The name is already too long:

![image](https://github.com/user-attachments/assets/7e23c35a-9c4e-42a8-81c1-cd50722f647f)


There is a slightly different behaviour for "Special tests" - when special tests fail in "canary" build, the constraints are still updated, and probably there are cases where we want them not to be updated - if for examp[le a new version of a dependency will make the OpenAPI tests to fail.

Reply:
ah i see, that make sense. Yeah thats good point, i think naming also well suites :) `CI image checks`

Reply:
If GitHub supported more workflow file calls, we could have a more flexible way to segregate workflows. However, they currently have a hard limit of 20 :)